---
title: "Proteomics, transcriptomics and radiomics data integration"
author: "Mafe Senosain"
date: "`r Sys.Date()`"
output:
    html_document:
        toc: true
        toc_float: true
editor_options: 
  markdown: 
    wrap: sentence
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning=FALSE)
```

```{r}
library(ComplexHeatmap)
library(circlize)
library(tidyverse)
library(dplyr)
library(RCy3)
library(igraph)
library(cluster)
library(factoextra)
library(plotly) 
library(umap) 
source('src/rnaseq/30_DEGanalysis.R')
environment_set()
```

```{r}
all_vars_scaled <- read.csv('data/TMA36_project/data_integration/cytof_rna_hm.csv', row.names = 1)
all_vars_raw <- read.csv('data/TMA36_project/data_integration/cytof_rna_hm_raw.csv', row.names = 1)
colnames(all_vars_scaled) = gsub('\\.', ' ', colnames(all_vars_scaled))
clusters_patients <- read.csv('data/TMA36_project/data_integration/clusters_patients.csv', row.names = 1)
clusters_features <- read.csv('data/TMA36_project/data_integration/clusters_features.csv', row.names = 1)
CDE = read.csv('data/TMA36_project/CDE/CDE_TMA36_2021SEPT21_DR_MF.csv')
CDE = CDE[match(rownames(all_vars_scaled), CDE$pt_ID),]
```

```{r}
# Add WES info to CDE
wes = read.csv("data/TMA36_project/WES/processed/wes_binary.csv")
colnames(wes)[1] = 'pt_ID'
CDE = left_join(CDE, wes, by = 'pt_ID')
# Add pt cluster to CDE
clusters_patients <- rownames_to_column(clusters_patients, var = 'pt_ID')
rownames(clusters_patients) <- clusters_patients$pt_ID
clusters_patients$pt_ID <- as.integer(clusters_patients$pt_ID)
CDE=left_join(CDE, clusters_patients, by = 'pt_ID')
```

```{r}
cde_mxif <- read.csv('data/TMA36_project/CDE/CDE_TMA36_2021SEPT21_DR_MF_forMxIF.csv')
#clusters_patients$pt_ID <- rownames(clusters_patients)
cde_mxif <- left_join(cde_mxif, clusters_patients, by = 'pt_ID')
write.csv(cde_mxif, file = 'data/TMA36_project/CDE/CDE_TMA36_2021SEPT21_DR_MF_forMxIF.csv', row.names = FALSE)
```

```{r}
sc_rna <- c('12929', '13774', '7984', '14965', '12889', '8356', 
            '11918', '11522', '15002', '15467', '13634', '14428',
            '11817', '14958', '12935', '13636')
CDE$scrna_data <- 'No'
CDE$scrna_data[which(CDE$pt_ID %in%  sc_rna)] <- 'Yes'
```

```{r}
# colors
col_prog = c('ind' = '#3498DB', 'int' = 'grey72', 'agg'= '#c75264')
col_sila = colorRamp2(c(0, 0.5, 1), c("white", "#6772c7", "#0f1547"))
col_stage = c('Stage 0'='#465362', 'Stage I'='#011936', 'Stage II' = '#c2eabd', 'Stage III' = '#f9dc5c', 'Stage IV' = '#ED254E')
col_hist = c('acinar'='#FFBF00', 'lepidic'='#f5ecda', 'micropapillary'='#2274A5', 'mucinous acinar'='#32936F', 'papillary'='#5C5C5C', 'solid'='#E83F6F')
col_smok = c('Current smoker'='#0FA3B1','Ex-smoker'='#B5E2FA','Never smoked'='#EDDEA4')
col_nodsiz = colorRamp2(c(0.5, 3, 8), c("white", "#ad61ac", "#70196f"))
col_sex = c('Female'='#1D2F6F', 'Male'='#8390FA')
col_age = colorRamp2(c(40, 65, 90), c("white", "#5BC0BE", "#3A506B"))
col_mutload = colorRamp2(c(0, 8), c("white", "#B32548"))
col_mut = c('0'= '#dbe2ff', '1'= '#0027c2')
#col_sc = c('Yes'=, 'No'=)

ha = HeatmapAnnotation(
  Prognosis = as.factor(CDE$n_op2),
  SILA = CDE$SILA,
  Stage = CDE$Stages_simplified,
  Histology = CDE$Hist_predominant,
  Smoking = CDE$Smoking_Status,
  Nodule_size = CDE$Path_Nodule_Size_cm,
  Sex = CDE$Gender,
  Age = CDE$Age_at_collection,
  #sc_rna = CDE$scrna_data,
  col = list(Prognosis = col_prog, 
             SILA = col_sila,
             Stage = col_stage,
             Histology = col_hist,
             Smoking = col_smok,
             Nodule_size = col_nodsiz,
             Sex = col_sex,
             Age = col_age
             ),
  gp = gpar(col = "black"),
  simple_anno_size = unit(0.2, "cm"),
  annotation_name_gp = gpar(fontsize = 8) #size of annotation name
)

bottom_ha = HeatmapAnnotation(
  Log_mut_load = log(CDE$mut_load),
  KRAS = as.factor(CDE$KRAS),
  RYR2 = as.factor(CDE$RYR2),
  MUC16 = as.factor(CDE$MUC16),
  EGFR = as.factor(CDE$EGFR),
  CTNND2 = as.factor(CDE$CTNND2),
  ADAMTS12 = as.factor(CDE$ADAMTS12),
  APOB = as.factor(CDE$APOB),
  NPAP1 = as.factor(CDE$NPAP1),
  PRDM9 = as.factor(CDE$PRDM9),
  SORCS1 = as.factor(CDE$SORCS1),
  ADGRB3 = as.factor(CDE$ADGRB3),
  DCHS2 = as.factor(CDE$DCHS2),
  LRRTM4 = as.factor(CDE$LRRTM4),
  PTPRC = as.factor(CDE$PTPRC),
  UNC13B = as.factor(CDE$UNC13B),
  CACNA1E = as.factor(CDE$CACNA1E),
  OBSCN = as.factor(CDE$OBSCN),
  col = list(
    Log_mut_load = col_mutload,
    KRAS = col_mut,
    RYR2 = col_mut,
    MUC16 = col_mut,
    EGFR = col_mut,
    CTNND2 = col_mut,
    ADAMTS12 = col_mut,
    APOB = col_mut,
    NPAP1 = col_mut,
    PRDM9 = col_mut,
    SORCS1 = col_mut,
    ADGRB3 = col_mut,
    DCHS2 = col_mut,
    LRRTM4 = col_mut,
    PTPRC = col_mut,
    UNC13B = col_mut,
    CACNA1E = col_mut,
    OBSCN = col_mut
  ),
  gp = gpar(col = "black"),
  simple_anno_size = unit(0.2, "cm"),
  annotation_name_gp = gpar(fontsize = 6), #size of annotation name
  show_legend = c(TRUE,TRUE,rep(FALSE, 16))
)
#row_ha = rowAnnotation(Subclust = as.factor(clusters_features2$cluster))

```

# Data integration: Heatmap
## Zoom out
```{r, fig.width = 11, fig.height = 7}
set.seed(0)
ht=Heatmap(t(all_vars_scaled), name = "z-score",
        heatmap_legend_param = list(color_bar = "continuous"), 
        row_names_gp = gpar(fontsize = 8),
        column_names_gp = gpar(fontsize = 8),
        row_split = clusters_features$cluster,
        show_row_dend = FALSE,
        column_split = clusters_patients$cluster,
        top_annotation = ha, show_row_names = F,
        bottom_annotation = bottom_ha
        )
padding = unit(c(2, 2, 2, 2), "mm")
draw(ht, heatmap_legend_side="right", padding=padding,
     #merge_legend = TRUE,
     annotation_legend_side = 'left')#, legend_grouping = "original")

```

## Zoom in
```{r, fig.width = 10, fig.height = 40}
set.seed(0)
Heatmap(t(all_vars_scaled), name = "z-score",
        heatmap_legend_param = list(color_bar = "continuous"), 
        row_names_gp = gpar(fontsize = 8),
        column_names_gp = gpar(fontsize = 8),
        row_split = clusters_features$cluster,
        column_split = clusters_patients$cluster,
        top_annotation = ha, show_row_names = T
        #left_annotation = row_ha
        )

```

# Data integration: PCA

## PCA features k=4
```{r, fig.width = 8, fig.height = 6}
library(PCAtools)
i_k = which(clusters_features$cluster ==1)
ii_k = which(clusters_features$cluster ==2)
iii_k = which(clusters_features$cluster ==3)
iv_k = which(clusters_features$cluster ==4)
clusters_features$cluster_let <- 'I'
clusters_features$cluster_let[ii_k] <- 'II'
clusters_features$cluster_let[iii_k] <- 'III'
clusters_features$cluster_let[iv_k] <- 'IV'
p <- pca(all_vars_scaled, metadata = clusters_features, center = F, scale = F)
PCAtools::biplot(p, x='PC1', y='PC2',
                 colkey = c('I' = '#2d7dd2', 'II' = '#97CC04', 'III'='#EEB902', 'IV'='#F45D01'),
                 colby='cluster_let', legendPosition='right',
                 colLegendTitle = 'Feature cluster', 
                 encircle = T, encircleFill = T,
                 lab = NULL,pointSize = 2
                 #showLoadings = TRUE, labSize = 2, sizeLoadingsNames = 5
                 )
```

## PCA patients k=4
```{r, fig.width = 8, fig.height = 6}
library(PCAtools)
clusters_patients$cluster <- as.factor(clusters_patients$cluster)
p <- pca(t(all_vars_scaled), metadata = clusters_patients, center = F, scale = F)
PCAtools::biplot(p, x='PC1', y='PC2',
                 colkey = c('1' = '#FF6666', '2' = '#96E000', '3'='#5D2E8C', '4'='#2EC4B6'),
                 colby='cluster', legendPosition='right',
                 colLegendTitle = 'Patient cluster', encircle = T, encircleFill = T,
                 lab = NULL,pointSize = 3)
```



# UMAP

## Features 3D
```{r}
#https://plotly.com/r/t-sne-and-umap-projections/
ft_umap = umap(t(all_vars_scaled), n_components = 3, random_state = 15) 
layout <- ft_umap[["layout"]] 
layout <- data.frame(layout) 
final <- cbind(layout, cluster=clusters_features$cluster, features=rownames(clusters_features)) 
col_pal <- RColorBrewer::brewer.pal(length(unique(final$cluster)), 'Dark2')

fig <- plot_ly(final, x = ~X1, y = ~X2, z = ~X3, color = ~final$cluster, colors = col_pal,
               hoverinfo='text',
               text = ~paste('</br> Cluster: ', cluster,
                             '</br> Feature: ', features)) %>% 
  add_markers(opacity = 0.5, marker = list(size=7)) %>% 
  layout(scene = list(xaxis = list(title = '0'), 
                                     yaxis = list(title = '1'), 
                                     zaxis = list(title = '2'))) 

fig
```

```{r}
ft_umap$layout
```


## Patients 3D
```{r}
ft_umap = umap(all_vars_scaled, n_components = 3, random_state = 15) 
layout <- ft_umap[["layout"]] 
layout <- data.frame(layout) 
final <- cbind(layout, cluster=clusters_patients$cluster, features=rownames(clusters_patients), CDE) 
col_pal <- RColorBrewer::brewer.pal(length(unique(final$cluster)), 'Dark2')

fig <- plot_ly(final, x = ~X1, y = ~X2, z = ~X3, color = ~final$cluster, colors = col_pal,
               hoverinfo='text',
               text = ~paste('</br> Cluster: ', cluster,
                             '</br> Feature: ', features,
                             '</br> SILA: ', SILA,
                             '</br> SILA_cat: ', n_op2,
                             '</br> Path stage: ', X8th_ed_path_stage)) %>% 
  add_markers(opacity = 0.5, marker = list(size=7)) %>% 
  layout(scene = list(xaxis = list(title = '0'), 
                                     yaxis = list(title = '1'), 
                                     zaxis = list(title = '2'))) 

fig
```



# Survival analysis

## All vs all
### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2', '3', '4'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('1', '2', '3', '4'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2', '3', '4'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('1', '2', '3', '4'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2', '3', '4'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('1', '2', '3', '4'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2', '3', '4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('1', '2', '3', '4'), legend_title = 'patient cluster')
```

## Survival 1 vs 2-3-4
```{r, fig.width = 4, fig.height = 5}
CDE$cluster_j <- CDE$cluster
CDE$cluster_j[CDE$cluster != '1'] <- '2-3-4'
```

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('1', '2-3-4'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('1', '2-3-4'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('1', '2-3-4'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('1', '2-3-4'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('1', '2-3-4'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('1', '2-3-4'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('1', '2-3-4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('1', '2-3-4'), legend_title = 'patient cluster')
```

## Survival 2 vs 1-3-4
```{r, fig.width = 4, fig.height = 5}
CDE$cluster_j <- CDE$cluster
CDE$cluster_j[CDE$cluster != '2'] <- '1-3-4'
```

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('2', '1-3-4'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('2', '1-3-4'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('2', '1-3-4'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('2', '1-3-4'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('2', '1-3-4'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('2', '1-3-4'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('2', '1-3-4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('2', '1-3-4'), legend_title = 'patient cluster')
```

## Survival 3 vs 1-2-4
```{r, fig.width = 4, fig.height = 5}
CDE$cluster_j <- CDE$cluster
CDE$cluster_j[CDE$cluster != '3'] <- '1-2-4'
```

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('3', '1-2-4'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('3', '1-2-4'), legend_title = 'patient cluster', rmst2_tau=3.5)
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('3', '1-2-4'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('3', '1-2-4'), legend_title = 'patient cluster', rmst2_tau=3.5)
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('3', '1-2-4'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('3', '1-2-4'), legend_title = 'patient cluster', rmst2_tau=3.5)
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('3', '1-2-4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('3', '1-2-4'), legend_title = 'patient cluster', rmst2_tau=3.5)
```

## Survival 4 vs 1-2-3
```{r, fig.width = 4, fig.height = 5}
CDE$cluster_j <- CDE$cluster
CDE$cluster_j[CDE$cluster != '4'] <- '1-2-3'
```

### Overall survival
```{r, fig.width = 4, fig.height = 5}
#colkey = c('1' = '#FF6666', '2' = '#96E000', '3'='#5D2E8C', '4'='#2EC4B6'),
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('4', '1-2-3'), 
              delete_group=FALSE, survival_type = 'OS', col_pal= c("#2EC4B6", "gray"),
              legend_labs = c('4', '1-2-3'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('4', '1-2-3'), 
              delete_group=FALSE, survival_type = 'RFS', col_pal= c("#2EC4B6", "gray"),
              legend_labs = c('4', '1-2-3'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('4', '1-2-3'), 
              delete_group=FALSE, survival_type = 'PFS', col_pal= c("#2EC4B6", "gray"),
              legend_labs = c('4', '1-2-3'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster_j', group_levels=c('4', '1-2-3'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('4', '1-2-3'), legend_title = 'patient cluster')
```

## Survival 1 vs 2

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('1', '2'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('1', '2'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('1', '2'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '2'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('1', '2'), legend_title = 'patient cluster')
```

## Survival 1 vs 3

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '3'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('1', '3'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '3'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('1', '3'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '3'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('1', '3'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '3'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('1', '3'), legend_title = 'patient cluster')
```

## Survival 1 vs 4

### Overall survival
```{r, fig.width = 4, fig.height = 5}
#colkey = c('1' = '#FF6666', '2' = '#96E000', '3'='#5D2E8C', '4'='#2EC4B6'),
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '4'), 
              delete_group=FALSE, survival_type = 'OS', col_pal = c('#FF6666', '#2EC4B6'),
              legend_labs = c('1', '4'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '4'), 
              delete_group=FALSE, survival_type = 'RFS', col_pal = c('#FF6666', '#2EC4B6'),
              legend_labs = c('1', '4'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '4'), 
              delete_group=FALSE, survival_type = 'PFS', col_pal = c('#FF6666', '#2EC4B6'),
              legend_labs = c('1', '4'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('1', '4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('1', '4'), legend_title = 'patient cluster')
```

## Survival 2 vs 3

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '3'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('2', '3'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '3'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('2', '3'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '3'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('2', '3'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '3'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('2', '3'), legend_title = 'patient cluster')
```

## Survival 2 vs 4

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '4'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('2', '4'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '4'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('2', '4'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '4'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('2', '4'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('2', '4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('2', '4'), legend_title = 'patient cluster')
```


## Survival 3 vs 4

### Overall survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('3', '4'), 
              delete_group=FALSE, survival_type = 'OS', 
              legend_labs = c('3', '4'), legend_title = 'patient cluster')
```

### RF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('3', '4'), 
              delete_group=FALSE, survival_type = 'RFS', 
              legend_labs = c('3', '4'), legend_title = 'patient cluster')
```

### PF survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('3', '4'), 
              delete_group=FALSE, survival_type = 'PFS', 
              legend_labs = c('3', '4'), legend_title = 'patient cluster')
```

### DRP survival
```{r, fig.width = 4, fig.height = 5}
x <- survival_plot(CDE, group_colname='cluster', group_levels=c('3', '4'), 
              delete_group=FALSE, survival_type = 'DRP', 
              legend_labs = c('3', '4'), legend_title = 'patient cluster')
```


